### ISSUE TITLE
[3.3.5] Spell Reflection vs Channel Spells, multiple reflections

### ISSUE THREAD

**Issue by Keader** [TrinityCore/TrinityCore#19440]
**Description:** Spell Reflection is reflecting some "ticks" of channel spells like Arcane Missiles, but not reflecting all ticks.

spell 42846 = [Arcane Missiles]
spell 23920 = [Spell Reflection]

**Expected behaviour:** Spell Reflection should reflect all ticks.

**Steps to reproduce the problem:**

1. Put a mage and warrior in duel.
2. With talent  [Missile Barrage] (spellid=44401) active, use Arcane Missiles on a warrior with Spell Reflection active.
3. You should receive back 2 Missiles, instead of 1.

It is easier to reproduce if the mage has a good haste rating.

**Branch(es):** 3.3.5

**TC rev. hash/commit:**  21b2042840e05ebeba365c54d47c07c13fde123f

**Comment by ariel-** [comment]
Did it ever work?

**Comment by Keader** [comment]
i dont think so, before your spells rework, all missiles hit warrior and 0 was reflected xd

**Comment by gmillerd** [comment]
Is improved spell reflection speced during this?

**Comment by ariel-** [comment]
Imp. spell reflection only affects party members

**Comment by MrSmite** [comment]
http://www.mmo-champion.com/threads/724175-Reflecting-channeled-casts?p=8076140&viewfull=1#post8076140

> in PVP, if you cast spell reflection BEFORE the warlocks cast starts:
- the three first waves of arcane missiles will get reflected, and the next ones you will have to take.
- the drain life cast will just simply cancel, and he will have to use another global cooldown to cast it again.
- the drain soul cast will simply cancel, and he will have to use another global cooldown to cast it again.
- the mind flay cast will simply cancel, and he will have to use another global cooldown to cast it again.

**Comment by ariel-** [comment]
Bad issue tagged in commit

**Comment by lineagedr** [comment]
@Keader you said
Expected behaviour: Spell Reflection should reflect the first "tick" only

Don't be stupid, @Keader Spell Reflection should reflect all of the missiles. Have you even even researched this?
https://youtu.be/Kat7dF9gPgc?t=18

**Comment by ariel-** [comment]
No need to insult other users, if you think something is wrong just state it.

**Comment by lineagedr** [comment]
Maybe if he researched it, I wouldn't have insulted him.

**Comment by Aokromes** [comment]
Ok, bad attitude even warned, next time you will be blocked.

**Comment by Keader** [comment]
```lineagedr->SetReact(REACT_PASSIVE);```  ![rolleyes](https://cloud.githubusercontent.com/assets/7120913/26352649/00731bb0-3f93-11e7-8527-40cd320bb6b6.gif)

Btw, i will try some friend to check it on retail, i'm pretty sure it only should reflect the first "tick"
I remember when i'm doing arena close to pre patch (wotlk->cata) and in middle of MOP my penance reflect only first tick.
With more time i will search more videos too :)

**Comment by Keader** [comment]
yeah i cant find videos to to sustain my point...
So updated issue description. Thanks lineagedr.
